### PR TITLE
Add high level package docs for rawdb & downloader

### PR DESCRIPTION
**Pull request by piersy**
This documentation is intended to simplify the task of getting a
holistic understanding of these packages.

### CODE REVIEW

**Comment by piersy** [comment]
Coverage from tests in `./e2e_test/...` for `./consensus/istanbul/...` at commit 3326c51ecc313b9335f53f6938ff0b17f73a435b
<details><summary>coverage:  49.7% of statements across all listed packages</summary><br>
<pre>
coverage:  61.2% of statements in consensus/istanbul
coverage:  40.0% of statements in consensus/istanbul/announce
coverage:  54.7% of statements in consensus/istanbul/backend
coverage:   0.0% of statements in consensus/istanbul/backend/backendtest
coverage:  24.3% of statements in consensus/istanbul/backend/internal/replica
coverage:  67.1% of statements in consensus/istanbul/core
coverage:  45.0% of statements in consensus/istanbul/db
coverage:   0.0% of statements in consensus/istanbul/proxy
coverage:  64.4% of statements in consensus/istanbul/uptime
coverage:  51.8% of statements in consensus/istanbul/validator
coverage:  79.2% of statements in consensus/istanbul/validator/random
</pre>
CommentID: 1c294f5967
</details>

File: eth/downloader/doc.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	current blocks. Setting ancient limit here enables "direct-ancient mode"
+	which I guess bypasses putting stuff into the main chain and then having it
+	be moved to the freezer later. I guess in full sync mode since all blocks
+	need to be processed all blocks need to go into the main database first and
+	only after they have been process can they be moved to the freezer, but
+	since fast sync does not process all blocks that step can be skipped.
+
+	Then, and I'm not really clear why if the origin is greater than the last
+	frozen block (ie there is stuff in the current database beyond whats in the
+	Freezer)the "direct-ancient mode is disabled", maybe because it is only
```
Comment (palango) [inline eth/downloader/doc.go:91]: missing space after ")"

File: eth/downloader/doc.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+sync.
+
+Light: Downloads only headers during sync and then downloads other data on
+demand in order to service rpc requests.
+
+Lightest: Like light but downloads only one header per epoch, wich on mainnet
+means one header out of every 17280 headers. This is particularly fast only
+takes 20 seconds or so to get synced.
+
+Sync process detail
```
[the remark this answers is not in the document]
Comment (piersy) [reply eth/downloader/doc.go:44]: Theres not any special formatting that you can apply for godoc, so I decided to leave it plain.

**Comment by piersy** [comment]
@palango Thanks for the comments, did the docs make sense to you?

**Review by palango — APPROVED** [review]
> did the docs make sense to you?

Yes, I think they give a good overview!

**Pull request merged**